File: gherkin/java/src/main/java/gherkin/Token.java
package gherkin;

import java.util.List;

public class Token {
    public final IGherkinLine line;
    public Parser.TokenType matchedType;
    public String matchedKeyword;
    public String matchedText;
    public List<GherkinLineSpan> mathcedItems;
    public int matchedIndent;
    public GherkinDialect matchedGherkinDialect;
    public Location location;

    public Token(IGherkinLine line, Location location) {
        this.line = line;
        this.location = location;
    }

    public boolean isEOF() {
        return line == null;
    }

    public void detach() {
        if (line != null)
            line.detach();
    }

    public String getTokenValue() {
        return isEOF() ? "EOF" : line.getLineText(-1);
    }

    @Override
    public String toString() {
        return String.format("%s: %s/%s", matchedType, matchedKeyword, matchedText);
    }
}

File: gherkin/dotnet/Gherkin/Token.cs
using Gherkin.Ast;

namespace Gherkin
{
    public class Token
    {
        public bool IsEOF { get { return Line == null; } }
        public IGherkinLine Line { get; set; }
        public TokenType MatchedType { get; set; }
        public string MatchedKeyword { get; set; }
        public string MatchedText { get; set; }
        public GherkinLineSpan[] MatchedItems { get; set; }
        public int MatchedIndent { get; set; }
        public GherkinDialect MatchedGherkinDialect { get; set; }
        public Ast.Location Location { get; set; }

        public Token(IGherkinLine line, Ast.Location location)
        {
            Line = line;
            Location = location;
        }

        public void Detach()
        {
            if (Line != null)
                Line.Detach();
        }

        public string GetTokenValue()
        {
            return IsEOF ? "EOF" : Line.GetLineText(-1);
        }

        public override string ToString()
        {
            return string.Format("{0}: {1}/{2}", MatchedType, MatchedKeyword, MatchedText);
        }
    }
}